pkg/daemon/update.go

```diff
@@ -270,6 +270,19 @@ func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) err
 			defer os.Remove(extensionsRepo)
 		}
 	}
+	changes := []string{}
+	if mcDiff.osUpdate {
+		changes = append(changes, "Upgrading OS")
+	}
+	if mcDiff.extensions {
+		changes = append(changes, "Installing extnesions")
+	}
+	if mcDiff.kernelType {
+		changes = append(changes, "Changing kernel type")
+	}
```

Comment:
nit: I'd factor out these ifs in another function (perhaps related to mcDiff itself, like `mdDiff.String()` or just a `.String()` method on the object to be directly formatted with `%s` when printing the object. Just returning `strings.Join` like below

Reply:
Sure, done!